tests/unit-pass/Makefile

```diff
@@ -1,60 +1,86 @@
-OS_TESTS = $(filter-out $(OS_EXCLUDES), $(filter-out $(EXCLUDES), $(filter-out %-link2.c, $(filter-out %-link2.C, ${OS_TUS}))))
-CPP_TESTS = $(filter-out %.c, ${TESTS})
-C_TESTS = $(filter-out %.C, ${TESTS})
-GCC_COMPILED_TESTS = ${TESTS:=.gcc}
-KCC_COMPILED_TESTS = ${TESTS:=.kcc}
-TEST_RESULTS = ${TESTS:=.out}
-REFERENCE_TEST_RESULTS = ${TESTS:=.ref}
-TEST_COMPARISON = ${TESTS:=.cmp}
-OS_TEST_COMPARISON = ${OS_TESTS:=.cmp}
-CPP_TEST_COMPARISON = ${CPP_TESTS:=.cmp}
-C_TEST_COMPARISON = ${C_TESTS:=.cmp}
+include ../check-results.mk
+
+TUS := $(wildcard ./*.c) $(wildcard ./*.C)
+EXCLUDES :=
+OS_TUS := $(wildcard ./j*.c) $(wildcard ./*.C)
+OS_EXCLUDES := ./stdio-scanf11.c \
+               ./stdio-scanf12.c \
+               ./stdio-scanf15.c \
+               ./stdio-bug3.c \
+               ./stdio-bug5.c \
+               ./stdio-bug6.c \
+               ./stdio-bug7.c \
+               ./stdio-bug12.c \
+               ./stdio-bug18.c \
+               ./stdio-bug19.c \
+               ./stdio-bug20.c \
+               ./stdio-fwrite.c \
+               ./stdio-temptest.c \
+               ./compcertTSO-2.c \
+               ./stdio-ferror.c \
+               ./stdio-fgets.c \
+               ./stdio-fseek.c \
+               ./stdio-scanf-flush.c \
+               ./stdio-ungetc.c \
+               ./stdio-sscanf.c \
+               ./stdio-fileno.c \
+               ./inst-check.c \
+               ./pthreads-starve.c \
+               ./fopen-enoent.c \
+               ./oldstyle-main.c \
+               ./struct-ini-1.c \
+               ./j016.c \
+               ./include-time.C \
+               ./new23.C \
+               ./new41.C
+TESTS := $(filter-out $(EXCLUDES), $(filter-out %-link2.c, $(filter-out %-link2.C, ${TUS})))
+OS_TESTS := $(filter-out $(OS_EXCLUDES), $(filter-out $(EXCLUDES), $(filter-out %-link2.c, $(filter-out %-link2.C, ${OS_TUS}))))
+CPP_TESTS := $(filter-out %.c, ${TESTS})
+C_TESTS := $(filter-out %.C, ${TESTS})
+GCC_COMPILED_TESTS := ${TESTS:=.gcc}
+KCC_COMPILED_TESTS := ${TESTS:=.kcc}
+TEST_RESULTS := ${TESTS:=.out}
+REFERENCE_TEST_RESULTS := ${TESTS:=.ref}
+TEST_COMPARISON := ${TESTS:=.cmp}
+OS_TEST_COMPARISON := ${OS_TESTS:=.cmp}
+CPP_TEST_COMPARISON := ${CPP_TESTS:=.cmp}
+C_TEST_COMPARISON := ${C_TESTS:=.cmp}
 
 ifeq ($(origin CC),default)
-	CC = gcc-7
+	CC := gcc-7
 endif
 
 ifeq ($(origin CXX),default)
-	CXX=g++-7
+	CXX := g++-7
 endif
 
 KCC := ../../dist/kcc
 KXX := ../../dist/k++
 
-LDFLAGS=-pthread
+LDFLAGS := -pthread
 
-CHECK_RESULT_COMPILE = if [ $$? -eq 0 ] ; then echo "passed $<"; mv $@.tmp.out $@.out; else echo "failed $<"; cat $@.tmp.out; exit 1; fi
-CHECK_RESULT_RUN = if [ $$? -eq 0 ] ; then echo "passed $<"; mv $@.tmp $@; else echo "failed $<"; cat $@.tmp; exit 1; fi
-CHECK_RESULT_RUN_VERBOSE = if [ $$? -eq 0 ] ; then echo "passed $<"; mv $@.tmp $@; else echo "failed $<"; cat $@.tmp; env VERBOSE=1 ./$<; exit 1; fi
 
-
-.PHONY: test clean reference comparison
```

Comment:
I really don't like this separate .PHONY declaration; this practice makes it is very likely to miss some targets -- which is indeed the case here. I put `.PHONY` on a per-target basis.